### PR TITLE
docs: Reword "Ordering guarantees" in Book

### PR DESCRIPTION
**Pull request by max-sixty**
Made a few changes:
- Tried to lead with what it means for the user
- Kept the theory but moved it down
- Tried to cut words where possible
- Changed the SQL example to one that DBs will actually get "wrong" rather than just an theoretical "SQL standard" one

### CODE REVIEW

File: web/book/src/transforms/sort.md [hunk 1]
```diff
@@ -42,39 +42,75 @@ sort [s"substr({first_name}, 2, 5)"]
...
+Here, PRQL pushes the `sort` down the pipeline, compiling the `ORDER BY` to the
+_end_ of the query. Consequently, most relation transforms retain the row order.
 
-For example, this query:
+The explicit semantics are:
+
+- `sort` introduces a new order,
+- `group` resets the order,
+- `join` maintains the order of the left relation,
+- database tables don't have a defined order.
```
Comment (aljazerzen) [inline web/book/src/transforms/sort.md]: Well, they do. Postgres will return the same order of rows, no matter how many times you execute a query (if you are not modifying the table in any way).

It's just that order is unknown.

(I'm a bit pedantic here, but people reading "semantics" are usually like that)

File: web/book/src/transforms/sort.md [hunk 2]
```diff
@@ -42,39 +42,75 @@ sort [s"substr({first_name}, 2, 5)"]
...
+
+Comparatively, in SQL, relations possess no order, being orderable solely within
+the context of the query result, CTE, or window functions. The lack of inherent
```
Comment (aljazerzen) [inline web/book/src/transforms/sort.md]:
```suggestion
the context of the query result, `TAKE` clause, or window functions. The lack of inherent
```

Relations of a CTEs do don't retain order. If they have ORDER BY, strictly speaking, it has no meaning.

**Review by aljazerzen — APPROVED** [review]
Much more readable!

File: web/book/src/transforms/sort.md (hunk 1 above)
Comment (max-sixty) [reply web/book/src/transforms/sort.md:61]: Pedantic is good here! I switched it

File: web/book/src/transforms/sort.md (hunk 2 above)
Comment (max-sixty) [reply web/book/src/transforms/sort.md]: This is "`LIMIT` statement"? Or it's something else?

File: web/book/src/transforms/sort.md (hunk 1 above)
Comment (max-sixty) [reply web/book/src/transforms/sort.md]: Pedantic is good here! I switched it

File: web/book/src/transforms/sort.md (hunk 2 above)
Comment (aljazerzen) [reply web/book/src/transforms/sort.md]: Oh yes of course, LIMIT, not take. Brainfart.

**Pull request merged**